src/app.js

```diff
@@ -60,12 +72,21 @@ function setupGame(){
         if(totalPlayers !== 1){
             console.log('Sorry, '+totalPlayers+' players are not supported yet');
         }else{
-            startGame(true);
+            setupMapMatrix()
         }
     });
 }
 
 function startGame(isNextTurn){
+    mapEngine.init();
+    map = mapEngine.map;
+
+    map[0][0] = {
+        aquiredBy : player,
+        type : "player",
+        explored : true
+    };
+    mapEngine.createMapFromArray(monsters.concat(bosses,specials,friends));
```

Comment:
@pedramp Unfortunately the map doesn't update with correct position of the player, this is probably because a new map is created every `startGame` call, I suggest you to move line 81-89 into line 51, i.e. under `setupGame` function and this will solve the issue.

Reply:
you're right